### PR TITLE
Feature/largest dimension first 2

### PR DESCRIPTION
**Pull request by redhug1**
### What

Add filter by largest dimension to create full sorted filter in function generateFullCSV()

### How to review

Study code and tests.
make test
make audit

### Who can review

Anyone

### CODE REVIEW

File: event/handler.go [hunk 1]
```diff
@@ -281,6 +283,55 @@ var SortFilter = func(ctx context.Context, handler *ExportHandler, event *Filter
...
+	dimensions, err := handler.datasetAPICli.GetVersionDimensions(ctx,
+		"", // userAuthToken
+		handler.serviceAuthToken,
+		"", // collectionID
+		event.DatasetID, event.Edition, event.Version)
+
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (ChrysmOre) [inline event/handler.go:304]: A couple of small formatting gripes, I think scopes should be closed on the same column they're open including for parenthesis when calling functions.. referring to the calls on lines 288 (above) and 300 (below). Also you probably don't want the blank line between creating and checking the error above.
Usually the first argument will start on the second line too unless you're opening a new scope in the arguments.

Comment (redhug1) [reply event/handler.go:304]: Thanks for review.
I suspect you meant line 308.
Gripes fixed.

File: cmd/dp-dataset-exporter/main.go [hunk 2]
```diff
@@ -32,11 +32,18 @@ import (
...
-	BuildTime string
+	//BuildTime string
 	// GitCommit represents the commit (SHA-1) hash of the service that is running
-	GitCommit string
+	//GitCommit string
 	// Version represents the version of the service that is running
-	Version string
+	//Version string
+
+	/* NOTE: replace the above with the below to run code with for example vscode debugger.*/
```
Comment (DavidSubiros) [inline cmd/dp-dataset-exporter/main.go]: I think we should uncomment `BuildTime`, `GitCommit` and `Version` above, and uncommet below?

File: event/handler.go [hunk 3]
```diff
@@ -281,6 +283,56 @@ var SortFilter = func(ctx context.Context, handler *ExportHandler, event *Filter
...
+	var dbFilterDimensions []*observation.Dimension
+
+	for _, dim := range dimensions.Items {
+		// Get the names of the options for a dimension
+		opts, err := handler.datasetAPICli.GetOptionsInBatches(ctx,
+			"", // userAuthToken
+			handler.serviceAuthToken,
+			"", // collectionID
+			event.DatasetID, event.Edition, event.Version, dim.Name,
+			100,
```
Comment (DavidSubiros) [inline event/handler.go:314]: maybe these values should be configurable? (or maybe we can do it later if we only want to test the performance difference at this point)

**Review by DavidSubiros — COMMENTED** [review]
- make test works
- make audit doesn't show any vulnerability
- Added a couple of small comments

File: cmd/dp-dataset-exporter/main.go (hunk 2 above)
Comment (redhug1) [reply cmd/dp-dataset-exporter/main.go]: ah, yes well spotted. Did no mean to book that in. now fixed

File: event/handler.go (hunk 3 above)
Comment (redhug1) [reply event/handler.go:314]: We only need to get the new feature working at this point.
Any further performance work will have to be a new ticket.

**Review by DavidSubiros — DISMISSED** [review]

**Review by DavidSubiros — APPROVED** [review]

**Pull request merged**